SimpleMachines/SMF
# Updates Unicode data files in a background task

Based on an idea that came up in [previous discussion](https://github.com/SimpleMachines/SMF/pull/7571#issuecomment-1288445293), this PR refactors the code from update_unicode_data.php into a background task so that SMF can update its Unicode data files itself.

### Pros:
- There is no longer any need to include potentially bulky amounts of changed Unicode data in SMF patch files.
- Forums can keep up to date with the latest version of Unicode automatically and without waiting for the next SMF patch release.
- Less work for us.

### Cons:
- ... 🤔 ... 🤔 ... 🤔 ...... nope, nothing comes to mind.

### Testing
I've tried everything I can think of to break this code, and it just keeps working no matter what I do. 
So now it is your turn! 😉

1. Checkout this PR (obviously).
2. In the admin panel, run the Weekly Maintenance scheduled task.
3. Wait for a minute or so in order to allow the background task plenty of time to complete.
4. Inspect the content of ./Sources/Unicode/Metadata.php to verify that SMF_UNICODE_VERSION has been updated to 15.1.0.0. 

### Notes
1. The update_unicode_data.php file still exists, but it now just impersonates cron.php in order to load and run the Update_Unicode background task on demand.
2. Although **_patches_** no longer need to include any updates to Unicode data, the **_install_** and **_upgrade_** packages should, so update_unicode_data.php should still be run as part of preparing a new release.
